music21/romanText/writeRoman.py

```diff
@@ -0,0 +1,397 @@
+                             '']  # One blank line between metadata and analysis
+        # Note: blank analyst and proof reader entries until supported within music21 metadata
+
+        self.currentKeyString: str = ''
+        self.prepSequentialListOfLines()
+
+    def prepTitle(self,
+                  md: metadata):
+        '''
+        Attempt to prepare a single work title from the score metadata looking at each of
+        the title, movementNumber and movementName attributes.
+        Failing that, a placeholder 'Unknown title' stands in.
+
+        >>> s = stream.Score()
+        >>> rnScore = romanText.writeRoman.RnWriter(s)
+        >>> rnScore.title
+        'Title unknown'
+
+        >>> s.insert(0, metadata.Metadata())
+        >>> s.metadata.title = 'Fake title'
+        >>> s.metadata.movementNumber = 123456789
+        >>> s.metadata.movementName = 'Fake movementName'
+        >>> rnScoreWithMD = romanText.writeRoman.RnWriter(s)
+        >>> rnScoreWithMD.title
+        'Fake title - No.123456789: Fake movementName'
+        '''
+
+        workingTitle = []
+
+        if md.title:
+            workingTitle.append(md.title)
+        if md.movementNumber:
+            workingTitle.append(f'- No.{md.movementNumber}:')  # Spaces later
+        if md.movementName:
+            if md.movementName != md.title:
+                workingTitle.append(md.movementName)
+
+        if len(workingTitle) > 0:
+            self.title = ' '.join(workingTitle)
+
+# ------------------------------------------------------------------------------
+
+    def prepSequentialListOfLines(self):
+        '''
+        Prepares a sequential list of text lines, with time signatures and Roman numerals
+        adding this to the (already prepared) metadata preamble ready for printing.
+
+        >>> p = stream.Part()
+        >>> m = stream.Measure()
+        >>> m.insert(0, meter.TimeSignature('4/4'))
+        >>> m.insert(0, roman.RomanNumeral('V', 'G'))
+        >>> p.insert(0, m)
+        >>> testCase = romanText.writeRoman.RnWriter(p)
+        >>> testCase.combinedList[-1]  # Last entry, after the metadata
+        'm0 b1 G: V'
+        '''
+
+        for thisMeasure in self.container.getElementsByClass('Measure'):
+            # TimeSignatures
+            tsThisMeasure = [t for t in thisMeasure.getElementsByClass('TimeSignature')]
+            if tsThisMeasure:
+                firstTS = tsThisMeasure[0]
+                self.combinedList.append(f'Time Signature: {firstTS.ratioString}')
+                if len(tsThisMeasure) > 1:
+                    unprocessedTSs = [x.ratioString for x in tsThisMeasure[1:]]
+                    msg = f'further time signature change(s) unprocessed: {unprocessedTSs}'
+                    self.combinedList.append(f'Note: {msg}')
+
+            # RomanNumerals
+            measureString = ''  # Clear for each measure
+            rnsThisMeasure = [r for r in thisMeasure.getElementsByClass('RomanNumeral')]
```

Comment:
It's subtle, but you'll see a performance improvement from not using a list comprehension. `getElementsByClass` returns a generator, so you can get right to this:
`rnsThisMeasure = thisMeasure.getElementsByClass('RomanNumeral')`

You'll save the memory overhead of storing the entire list as well as the generator overhead from iterating an extra time: once to apply `r for r` and once to do `for rn in myListComprehension`

FYI @mscuthbert there's a pylint check we could consider turning back on that checks for unnecessary list comprehensions.

Reply:
I think it's turned off because there is one side effect of iterating over a generator -- all the RNs in thisMeasure now have their activeSites set to thisMeasure so if later you do, thisMeasure._elements[20] it will have the activeSite set.  BUT we're definitely not doing it here, and probably we should have the pylint setting turned on and just disable it the one or two places that it is actually not unnecessary.

There's another pylint thing with statement-has-no-effect, that I think I had disabled because of how filters used to work that pylint could not detect, that I think I can turn on.

There are a bunch of pylint settings that I want to turn back on, BUT haven't had time to refactor enough places to do it.  Mostly relating to statement complexity, etc. -- I really want to encourage code that my students can read.